src/sinks/datadog/traces/stats.rs

```diff
@@ -0,0 +1,532 @@
+use std::collections::BTreeMap;
+
+use chrono::Utc;
+use prost::Message;
+use serde::{Deserialize, Serialize};
+use serde_bytes;
+
+use crate::{
+    event::{TraceEvent, Value},
+    metrics::AgentDDSketch,
+    sinks::datadog::traces::sink::PartitionKey,
+};
+
+mod ddsketch_full {
+    include!(concat!(env!("OUT_DIR"), "/ddsketch_full.rs"));
+}
+
+const TOP_LEVEL_KEY: &str = "_top_level";
+const SAMPLING_RATE_KEY: &str = "_sample_rate";
+const MEASURED_KEY: &str = "_dd.measured";
+const PARTIAL_VERSION_KEY: &str = "_dd.partial_version";
+
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+struct AggregationKey {
+    payload_key: PayloadAggregationKey,
+    bucket_key: BucketAggregationKey,
+}
+
+impl AggregationKey {
+    fn new_aggregation_from_span(
+        span: &BTreeMap<String, Value>,
+        payload_key: PayloadAggregationKey,
+    ) -> Self {
+        AggregationKey {
+            payload_key: payload_key.with_span_context(span),
+            bucket_key: BucketAggregationKey {
+                service: span
+                    .get("service")
+                    .map(|v| v.to_string_lossy())
+                    .unwrap_or_else(|| "".into()),
```

Comment:
Is having `""` strings here regular operation, or is it worth putting that information into the internal log?

Reply:
The only concern I would have would be when we would have trace coming from a non-datadog source that may come without a service field on purpose, but this is highly speculative.

Reply:
This feels reasonable - or perhaps going further allowing users to set this in their config (perhaps template-able). Though that could be an enhancement down the line based on feedback.